controllers/component_build_controller.go

```diff
@@ -182,18 +215,38 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		// Deletion of the component is requested
 
 		if controllerutil.ContainsFinalizer(&component, ImageRegistrySecretLinkFinalizer) {
-			if err := r.removeImageRegistrySecretLink(ctx, &component); err != nil {
+			pipelineSA := &corev1.ServiceAccount{}
+			err := r.Client.Get(ctx, types.NamespacedName{Name: "pipeline", Namespace: req.Namespace}, pipelineSA)
+			if err != nil && !errors.IsNotFound(err) {
+				log.Error(err, fmt.Sprintf("Failed to read service account %s in namespace %s", buildPipelineServiceAccountName, req.Namespace))
 				return ctrl.Result{}, err
 			}
+			if err == nil { // If pipeline service account found, unlink the secret from it
+				imageRepoGenerated, imageRepoSecretName, err := getComponentImageRepoAndSecretNameFromImageAnnotation(&component)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+				if imageRepoGenerated != getContainerImageRepository(component.Spec.ContainerImage) {
+					// User defined repository was used
+					imageRepoSecretName = imageRegistryUserSecretName
```

Comment:
If user does not specify ContainerImage and image-controller generates the image repo, when execute to here, isn't `component.Spec.ContainerImage` an empty string? How could this case happen?